### ISSUE TITLE
Working with distinguished names

### ISSUE THREAD

**Issue by cajus** [cannatag/ldap3#188]
It looks like the "old" python-ldap module is no more shipped by some distributions, so I'm currently looking for something I can migrate my code base to. Just found the ldap3 module - and here I am ;-)

Currently I'm looking into the ldap3.utils.dn methods, and I either don't understand how they're intended to be used or there's a bit missing somewhere.

My old code contains parts needed to split the DN into its main parts (including the RDN) and do something with it. Then assemble it back to its string representation.

Simple example that gets the container DN for a given DN:

```
from ldap.dn import dn2str, str2dn
dn = "cn=Frank Möller+uid=moellerm,ou=sales,dc=example,dc=com"
print(dn2str(str2dn(dn)[1:]))
```

How can I achieve this with the ldap3 module? There's a `parse_dn` that outputs some format which is complicated to split (because of the RDN "+" in the tuple where I've to inspect every single item if it's part of the previous RDN or not) and there doesn't seem to be a method in the ldap3.utils.dn namespace which can reverse the process of parsing to assemble a DN string.

**Comment by cannatag** [comment]
Hi,
the ldap3.utils.dn package is still not documented. Tou should use the ``to_dn()`` function:
```
from ldap3.utils.dn import to_dn
dn = "cn=Frank Möller+uid=moellerm,ou=sales,dc=example,dc=com"
print(','.join(to_dn(dn)[1:]))
```

should work the same.

**Comment by cajus** [comment]
Hmm. There seems to be some additional print output if I do that. It is rendered like this:

```
>>> to_dn(dn)
c
n
=
F
r
a
n
k
 
M
ö
l
l
e
r
+
u
i
d
=
m
o
e
l
l
e
r
m
o
u
=
s
a
l
e
s
d
c
=
e
x
a
m
p
l
e
d
c
=
c
o
m
['cn=Frank Möller+uid=moellerm', 'ou=sales', 'dc=example', 'dc=com']
>>> 
```

**Comment by cannatag** [comment]
Hi, I'm not able to replicate that, which version of ldap3 are you using? Which version of Python? On what platform?

**Comment by cajus** [comment]
I'm using python3.5 on Fedora 24 and just did a "pip install ldap3" inside of a virtualenv. From the installed eggs it looks like version 1.3.1.

Here's a "print(c)" in the dn.py. I can't find it in git, though. Maybe I've added it some time ago to try to understand how these methods are intended to use. I don't remember and it's strange because I clean the virtualenv from time to time. Whatever. I've removed the print() and - what a wonder - the output is gone away...

Currently I found that porting pyldap to debian/ubuntu is less work than adapting hundreds of lines of LDAP code to the ldap3 module due to usage incompatibilities. So I'll stay there for now. Maybe I'll do some API glue layer if there's a need to migrate. Whatever.

Thanks for your time, anyway.
